lib/modules/platform/scmm/index.ts

```diff
@@ -0,0 +1,349 @@
+import { logger } from '../../../logger';
+import type { BranchStatus } from '../../../types';
+import * as git from '../../../util/git';
+import * as hostRules from '../../../util/host-rules';
+import { sanitize } from '../../../util/sanitize';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfigByContent,
+  EnsureCommentRemovalConfigByTopic,
+  EnsureIssueConfig,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { repoFingerprint } from '../util';
+import { smartTruncate } from '../utils/pr-body';
+import { mapPrFromScmToRenovate } from './mapper';
+import ScmClient from './scm-client';
+import { getRepoUrl, mapPrState, matchPrState, smartLinks } from './utils';
+
+interface SCMMRepoConfig {
+  repository: string;
+  prList: Pr[] | null;
+  defaultBranch: string;
+}
+
+export const id = 'scmm';
+
+let config: SCMMRepoConfig = {} as any;
+let scmmClient: ScmClient;
+
+export async function initPlatform({
+  endpoint,
+  token,
+}: PlatformParams): Promise<PlatformResult> {
+  if (!endpoint) {
+    throw new Error('SCM-Manager endpoint not configured');
+  }
+
+  if (!token) {
+    throw new Error('SCM-Manager api token not configured');
+  }
+
+  scmmClient = new ScmClient(endpoint, token);
+
+  const me = await scmmClient.getCurrentUser();
+  const gitAuthor = `${me.displayName} <${me.mail}>`;
+  const result = { endpoint, gitAuthor };
+
+  logger.info(`Plattform initialized ${JSON.stringify(result)}`);
+
+  return result;
+}
+
+export async function initRepo({
+  repository,
+  gitUrl,
+}: RepoParams): Promise<RepoResult> {
+  const repo = await scmmClient.getRepo(repository);
+  const defaultBranch = await scmmClient.getDefaultBranch(repo);
+  const url = getRepoUrl(
+    repo,
+    gitUrl,
+    /* istanbul ignore next */
+    hostRules.find({ hostType: id, url: scmmClient.getEndpoint() }).username ??
+      '',
+    process.env.RENOVATE_TOKEN ?? '',
+  );
+
+  config = {} as any;
+  config.repository = repository;
+  config.defaultBranch = defaultBranch;
+
+  await git.initRepo({
+    ...config,
+    url,
+  });
+
+  // Reset cached resources
+  invalidatePrCache();
+
+  const result = {
+    defaultBranch: config.defaultBranch,
+    isFork: false,
```

Comment:
remove

Reply:
I am sorry, i don't understand, what should be removed and why. Could you please explain it to me?

Reply:
But the isFork property is not an optional property. It can not be undefined. 

```ts
export interface RepoResult {
  defaultBranch: string;
  isFork: boolean;
  repoFingerprint: string;
}
```